### PR TITLE
[WIP]support proxy in maven plugins(exclude asc)

### PR DESCRIPTION
**Pull request by andxu**
Fix issue: https://github.com/microsoft/azure-maven-plugins/issues/844

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/util/ValidationUtil.java [hunk 1]
```diff
@@ -41,17 +41,18 @@ public static AuthConfiguration validateMavenAuthConfiguration(AuthConfiguration
     }
 
     public static void validateHttpProxy(String httpProxyHost, String httpProxyPort) throws InvalidConfigurationException {
-        if ((StringUtils.isNotBlank(httpProxyHost) && StringUtils.isBlank(httpProxyPort)) ||
-                (StringUtils.isBlank(httpProxyHost) && StringUtils.isNotBlank(httpProxyPort))) {
+        if (StringUtils.isAllBlank(httpProxyHost, httpProxyPort)) {
+            return;
+        }
+        if (!StringUtils.isNoneBlank(httpProxyHost, httpProxyPort)) {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/util/ValidationUtil.java:47]: May use `isAnyBlank(String...)` here

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AzureAuthManager.java [hunk 2]
```diff
@@ -66,7 +71,11 @@
         }
         return retrievers.retrieve().onErrorResumeNext(e ->
                 Single.error(new LoginFailureException(String.format("Cannot get credentials from authType '%s' due to error: %s", authType, e.getMessage())))
-        );
+        ).doOnSuccess(credentialWrapper -> {
+            if (StringUtils.isNotBlank(configuration.getHttpProxyHost())) {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AzureAuthManager.java:75]: Do we need to set proxy before retrieve as we need to validate the token which may need network access

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AzureAuthManager.java:75]: seems we needn't to save proxy back to CredentialWrapper

**Comment by andxu** [comment]
replaced with https://github.com/microsoft/azure-maven-plugins/pull/1240

**Pull request closed without merging**